test/test_jit_disabled.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+import sys
+import os
+import contextlib
+import subprocess
+
+
+@contextlib.contextmanager
+def _jit_disabled():
+    cur_env = os.environ.get("PYTORCH_JIT", "1")
+    os.environ["PYTORCH_JIT"] = "0"
+    try:
+        yield
+    finally:
+        os.environ["PYTORCH_JIT"] = cur_env
+
+_program_string = """
+import torch
+class Foo(torch.jit.ScriptModule):
+    def __init__(self, x):
+        super(Foo, self).__init__()
+        self.x = torch.jit.Attribute(x, torch.Tensor)
+
+    def forward(self, input):
+        return input
+
+s = Foo(torch.ones(2, 3))
+print(s.x)
+"""
+
+class TestJitDisabled(unittest.TestCase):
+    """
+    These tests are separate from the rest of the JIT tests because we need
+    run a new subprocess and `import torch` with the correct environment
+    variables set.
+    """
+    def test_attribute(self):
+        with _jit_disabled():
+            out_disabled = subprocess.check_output([
```

Comment:
Instead of hardcoding this in every test method, can you use the `--subprocess` to run each method in subprocess 
https://github.com/pytorch/pytorch/blob/c002ede1075d05ab82e1d50fcc5f94ec1e0d95a9/test/common_utils.py#L50
and modify `run_test.py` to add a special handler for `test_jit_disabled` similar to
https://github.com/pytorch/pytorch/blob/c002ede1075d05ab82e1d50fcc5f94ec1e0d95a9/test/run_test.py#L113-L114?

Reply:
And the `setUp` method in `test_jit_disabled.py` should assert that the env flag is correctly set and pytorch jit is disabled.

Reply:
The environment variable has to be set before the test is run since it's checked when `import torch` happens, and it doesn't look like the existing subprocess stuff can handle that. Maybe something like a custom runner for this class like would work:

```python
class TestJitDisabled(JitTestCase):
    def run(self, result=None):
        with _jit_disabled():
			super(TestJitDisabled, self).run(result)
```